From c062504a2864680fee9c81ea46e7613315de038a Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Mon, 15 Jun 2020 11:24:32 +0200
Subject: [PATCH] Add missing Qt5WebChannel dependency

QWebChannel header is used in src/lib/marble/MarbleLegendBrowser.cpp
since commit b42ee5c8 but apparently not always implicitly included.

Downstream report: https://bugs.gentoo.org/728234

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt                | 1 +
 src/lib/marble/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 631a5b7d7..a3b2139dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,6 +126,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL Android)
 
    set ( MARBLE_NO_WEBKITWIDGETS TRUE )
 else()
+    find_package(Qt5WebChannel ${REQUIRED_QT_VERSION})
     find_package(Qt5WebEngine ${REQUIRED_QT_VERSION})
     find_package(Qt5WebEngineWidgets ${REQUIRED_QT_VERSION})
 
diff --git a/src/lib/marble/CMakeLists.txt b/src/lib/marble/CMakeLists.txt
index e02c7d160..7d27be94b 100644
--- a/src/lib/marble/CMakeLists.txt
+++ b/src/lib/marble/CMakeLists.txt
@@ -395,6 +395,7 @@ target_link_libraries(marblewidget
 if (NOT MARBLE_NO_WEBKITWIDGETS)
     target_link_libraries(marblewidget
         LINK_PUBLIC
+            Qt5::WebChannel
             Qt5::WebEngineWidgets
             Qt5::WebEngine
     )
-- 
2.27.0

